src/charm.py

```diff
@@ -250,6 +265,37 @@ def _service_ports(self) -> List[ServicePort]:
             ServicePort(int(port), name=name) for name, port in self._tcp_entrypoints().items()
         ]
 
+    @property
+    def _forward_auth_config(self) -> ForwardAuthRequirerConfig:
+        ingress_app_names = [
+            rel.app.name  # type: ignore
+            for rel in itertools.chain(
+                self.ingress_per_appv1.relations,
+                self.ingress_per_appv2.relations,
+                self.ingress_per_unit.relations,
+                self.traefik_route.relations,
+            )
+        ]
+        return ForwardAuthRequirerConfig(ingress_app_names)
+
+    @property
+    def _is_forward_auth_enabled(self) -> bool:
+        if self.config["enable_experimental_forward_auth"]:
+            return True
+        return False
+
+    def _on_forward_auth_config_changed(self, event: AuthConfigChangedEvent):
+        if self._is_forward_auth_enabled:
+            if self.forward_auth.is_ready():
+                self._process_status_and_configurations()
```

Comment:
**Question**: In case `is_ready` is False during this hook, it will be addressed on update-statue the latest. Is that ok?

Would be useful to have this situation captured in a unit test.

Reply:
As mentioned, we rely on `update-status` for edge cases. Do you think we should defer the event instead?

Reply:
We should avoid both 😅 
But I see that `self._process_status_and_configurations()` is traefik's "common exit hook" so no issue here at all!